base_user_role/tests/test_user_role.py

```diff
@@ -3,51 +3,55 @@
 import datetime
 
 from odoo import fields
-from odoo.tests.common import TransactionCase
+from odoo.tests.common import SavepointCase
```

Comment:
`SavepointCase` has been deprecated in 15.0 and is replaced by `TransactionCase`: https://github.com/odoo/odoo/blob/15.0/odoo/tests/common.py#L746-L753=

Outside of this, ok to switch `setUp` to `setUpClass` but I can't see other fix, which issue you encountered exactly?

Reply:
@sebalix when accessing the "Role" menu, the `base.res_users_rule` was triggered and an error was raised (because field `user_ids` might contain users with companies that are not chosen by the current user in the proper widget). I've added a proper test for it and force-pushed it a few seconds ago.